Jenkinsfile

```diff
@@ -0,0 +1,66 @@
+import groovy.json.JsonOutput
+
+pipeline {
+    agent any
+    stages {
+        stage("Create Databricks cluster") {
+            steps {
+                script {
+                    def uuid = UUID.randomUUID().toString()
+                    def clusterParams = [
+                        cluster_name: "jenkins-${uuid}",
+                        spark_version: "6.2.x-scala2.11",
+                        node_type_id: "i3.xlarge",
+                        num_workers: 1,
+                        autotermination_minutes: 10,
+                    ]
+                    def clusterParamsJson = JsonOutput.toJson(clusterParams)
+
+                    def createClusterCmd = """databricks clusters create --json '${clusterParamsJson}'"""
+                    createClusterOutput = sh(script: createClusterCmd, returnStdout: true)
+                    echo createClusterOutput
+
+                    def createClusterJson = readJSON(text: createClusterOutput)
+                    // No def makes it a global variable, accessible in other stages
+                    clusterId = createClusterJson["cluster_id"]
+                }
+            }
+        }
+        stage("Set up Databricks Connect") {
+            steps {
+                script {
+                    withCredentials([
+                        string(credentialsId: 'databricks-connect-api-token', variable: 'API_TOKEN'),
+                        string(credentialsId: 'databricks-workspace-url', variable: 'WORKSPACE_URL')
+                    ]) {
+                        def dbConnectParams = [
+                            host: WORKSPACE_URL,
+                            token: API_TOKEN,
+                            cluster_id: clusterId,
+                            org_id: "0",
+                            port: "15001",
+                        ]
+                        def dbConnectParamsJson = JsonOutput.toJson(dbConnectParams)
+                        sh "echo '$dbConnectParamsJson' > ~/.databricks-connect"
+
+                        // Smoke test to check if databricks-connect is set up correctly
+                        def sparkHome = "/usr/lib/python3.7/site-packages/pyspark"
+                        sh "SPARK_HOME=${sparkHome} databricks-connect test"
+                    }
+                }
+            }
+        }
+        stage("Run tests") {
+            steps {
+                echo "TODO 2"
+            }
+        }
+        stage("Terminate Databricks cluster") {
```

Comment:
Is there a way to run this stage regardless of success or failure of previous stages? I know we are setting auto-termination timeout.

Reply:
Sure good idea. One minor caveat will be that it will attempt to terminate the cluster even if it wasn't successfully created in the first place, but that's fine.